### ISSUE TITLE
An alternative strategy to moderation

### ISSUE THREAD

**Issue by MylesBorins** [nodejs/community-committee#122]
Copied from https://github.com/nodejs/TSC/issues/114

---

So I've been thinking quite a bit about moderation, and specifically some of the pain points that we have been dealing with. The biggest problems that I currently see are. You will find bold TLDR's throughout this brain dump of the key points.

1. It is impossible to codify code-of-conduct violations without marginalizing a group
   - a good example is the current state of cursing and sexualized language.
   - The discussion around coming up with what is appropriate here results in massive derailment and divide in our internal community

2. Banning is a big deal, thus we debate quite a bit about banning individuals
   - While we are busy discussing things people are being harassed
   - When we decide not to act we set a precedent that certain behavior is ok, and this can and has escalated in the past.

TLDR; **we should not be codifying specific acts but rather rely on individual contributors reaction on a case by case basis and we should be more liberal with our banning. The most important thing is to make a safe space for individuals to contribute. The second most important thing is to avoid derailment.**

The project is made up of individuals from all walks of life. We all come with different value systems, philosophies, and agendas. Trying to codify rules or guidelines will only serve to make those differences clearer, it will only divide us. An alternative is to make use of our differences. Does it matter what we have codified if a contributor is offended, marginalized, or feels silenced? With this in mind I would like to propose:

A specific code of conduct violation should be at the discretion of the individual making the moderation request. We are a group of trusted collaborators. If someone from the group feels specific behavior is inappropriate, it is safe to assume that other individuals in the community will feel the same way. If the goal is to create a safe space, then it shouldn't matter if we have check boxes for all the potential violations. Individuals are great at setting a line that shouldn't be crossed for them, and we should empower them to act on it. Spending our time defining that line is counter productive. We inherently trust our collaborators and members to be good actors, and such we should trust them to make the right decision in a situation where they feel the line has been crossed.  If a individual is abusing this privilege it would obviously cross the line of others and be dealt with. It is really easy to argue about a specific hard and fast rule, it is much harder to argue with an individuals experience.

TLDR; **A specific code of conduct violation should be at the discretion of the individual making the moderation request. Individuals are great at setting a line that shouldn't be crossed for them, and we should empower them to act on it**

This now brings up the act of banning. Currently banning has been reserved for repeat offenders or obvious bad actors. Generally if someone is banned it is a very big deal, and they are not very likely to have that ban removed (I can only think of one instance where a ban was lifted). I would like to propose an alternative banning approach closer to the way it is handled on IRC. In many IRC channels if you step over the line or derail the room you are quickly banned (at the discretion of the OP). Generally this is not a permaban... sometimes they are short bans to give someone a chance to calm down, other times there is a repeal process that is documented. Generally pm'ing one of the ops, stating that you understand what you did and won't do it again results in the ban being lifted. This is a fantastic way to set a precedent and stop derailment. A great example where this could have been useful was https://github.com/nodejs/node/pull/5790. Now I think that the way it was handled was fine, but a quick ban followed with an email explaining how to get unbanned would have sent a very clear message regarding derailment. The biggest win for this would be taking banning of of a pedestal. We are all going to make mistakes, taking away the power of a "ban" is a great way to make it way safer to make mistakes without permanent consequences.

TLDR; **We should be very quick to ban individuals for CoC violations or derailment. It should also be very quick to get unbanned by following a clear process. Only repeat offenders will be truly perma banned.**

At this point I want to make it clear that these ideas are still being thought through, and likely have some flaws. The goals I have here are to improve our moderation, make node a safe space, and to decrease derailment (especially internally). The biggest win for me would be if we could find a way to have a moderation policy that does not marginalize our collaborators and members. We spend too much time in fighting about semantics. We draw lines between philosophies and end up creating "others" within our own midst. We all care for node and want to see it succeed. As I can tell those that are investing time have the best intentions for the project. The policies listed above are primarily to help us screen out those who do not have the best intentions for our project / community, not punish those who do.

TLDR; **We all make mistakes. If it isn't such a big deal to be reprimanded for making a mistake, people will feel safer to make them.**

**Comment by bnb** [comment]
I am generally +1 to this - seems like a better policy than we've currently got in place that isn't really working (at no fault of the people working to follow and enforce it, mind you).

This part really spoke to me: 

>We are a group of trusted collaborators. If someone from the group feels specific behavior is inappropriate, it is safe to assume that other individuals in the community will feel the same way.

This touches on so many aspects of the moderation issues we've faced, and I believe is entirely accurate and fair. Thank you, Myles, for the extremely concise wording - I don't think it could have been written more accurately.

Some tasks that I see arising from it:
* How do we actually temp ban someone?
* What are ban periods, and how are they defined? **Where** and **who** defines these?
* Where does the documentation around the ban and revoking it land?

Finally, this is probably the most worrisome and sensitive part of this discussion to me:
>If a individual is abusing this privilege it would obviously cross the line of others and be dealt with.

How do we define "abusing this privilege" and more importantly how is it dealt with. This sentence glazes over this issue a bit, but I believe having a strong discourse around this with explicitly defined goals and actions (leave _nothing_ to question) agreed to by the TSC and CommComm is the only true possibility of a blocker to this landing.

**Comment by mcollina** [comment]
> How do we define "abusing this privilege" and more importantly how is it dealt with. This sentence glazes over this issue a bit, but I believe having a strong discourse around this with explicitly defined goals and actions (leave nothing to question) agreed to by the TSC and CommComm is the only true possibility of a blocker to this landing.

This is important. IMHO the problem is not about making mistakes, but rather making mistakes to purse one's agenda. A mistake can be quickly amended if that happens.

**Comment by mcollina** [comment]
Definitely 👍  to this proposal.

**Comment by mhdawson** [comment]
I think this is a great starting point for discussion by the new members of the moderation team.

**Comment by gibfahn** [comment]
>While we are busy discussing things people are being harassed
>We should be very quick to ban individuals for CoC violations or derailment.

Agreed on both points.

I think the problem is (personal opinion) that we don't have a clear process that empowers collaborators to act. We have [the moderation policy](https://github.com/nodejs/TSC/blob/master/Moderation-Policy.md), but it's long and leaves quite a lot up to the moderation team's discretion. 

How about this as a proposal:

#### Moderation process:

- When a collaborator wants to take a moderation action<sup>1</sup> they quote it in an issue in the moderation repo (or a PR in an `admin` repo if we use OrgBot - see https://github.com/nodejs/community-committee/issues/22).
- As soon as one other collaborator agrees (assuming no objections<sup>2</sup>) they should go ahead and moderate.

<sup>1</sup> It might be better to just ask the user to moderate their comment first, depending on the situation (whether you think they're acting in good faith). That doesn't need approvals.
<sup>2</sup> If another collaborator has replied/engaged then that might be a sign that they think the comment shouldn't be moderated, asking them to confirm seems reasonable.

#### Default moderation actions:

1. For a single offence just remove the comment (or use a `<details>` block).
2. If it's a repeat offence ban for a week (if we think they're a bot then skip this and ban indefinitely).
3. If they've been banned for a week and again then offend again ban indefinitely. Indefinite ban comes with a stock email explaining what they were banned for and how to get unbanned.

That way we can very quickly get to a temp-ban (user commented twice in a way that you find offensive, screenshots, issue in Moderation, one 👍 , temp-ban). Then we have a week's time to discuss, and the user has a week's time to cool off.

Having an email describing a process for getting unbanned seems important to me, it seems like an easy way to filter out the majority who'll just move on from the people who actually want to continue to be part of the community. 

As an aside, I'd say the fact that only org owners have the power to ban people is one of the reasons we are slow to ban. Having a bot do it seems like a good solution to that problem.

**Comment by bnb** [comment]
@MylesBorins is this something we should still be interested in working on? Doing some cleanup and want to check in on the status of this + see if there's anything actionable moving forward for the CommComm or Moderation Team to do.
